src/lib/Libnet/net_server.c

```diff
@@ -528,6 +560,28 @@ wait_request(time_t waittime)
 			return (-1);
 		}
 	} else {
+		if (scks) {
+			fd_set		fdset;
+			struct timeval  timeout;
+			timeout.tv_usec = 0;
+			timeout.tv_sec = 0;
+			FD_ZERO(&fdset);
+                        for (i = 0; i < scks->active_socks; i++) {
+                                if (scks->socket_fd[i] != 0) {
+					FD_SET(scks->socket_fd[i], &fdset);
+				}
+			}
+			if (select(FD_SETSIZE, &fdset, NULL, NULL, &timeout) != -1) {
+				 for (i = 0; i < scks->active_socks; i++) {
+				 	if (FD_ISSET(scks->socket_fd[i], &fdset)) {
+						log_event(PBSEVENT_DEBUG3, PBS_EVENTCLASS_SERVER,
+							LOG_DEBUG, __func__, "processing priority sockets");
+						return process_socket(scks->socket_fd[i]);
```

Comment:
We need to return from this function after processing all the priority sockets. But the current code returns as soon as it processes only one socket from the list of priority sockets. Please change the code accordingly.

Reply:
done